### PR TITLE
Fix abs diff computation in check_batch test utility

### PR DESCRIPTION
**Pull request by stiepan**
## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
The `check_batch` utility reports wrong abs differences for unsigned types if the actual difference is bigger than the half of the range of the type. (Take uint8 and 1 vs 254 - the abs diff is 253 while the chekc_batch would report 3).

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by stiepan** [comment]
!build

**Comment by stiepan** [comment]
!build

File: dali/test/python/test_utils.py [hunk 1]
```diff
@@ -128,6 +129,51 @@ def get_gpu_num():
...
+
+    # np.abs of diff doesn't handle overflow for unsigned types
+    absdiff = np.maximum(left, right) - np.minimum(left, right)
+    # max - min can overflow for signed types, wrap them up
+    absdiff = absdiff.astype(make_signed(absdiff.dtype))
+    return absdiff
+
+
+def _check_absdiff():
+    for i in range(-128, 127):
```
Comment (stiepan) [inline dali/test/python/test_utils.py:152]: In principle, overflow on signed int is UB (that we rely on anyway without this PR). The following one-time check aims to verify it works as expected.

**Review by awolant — APPROVED** [review]

File: dali/test/python/test_utils.py [hunk 2]
```diff
@@ -128,6 +129,51 @@ def get_gpu_num():
     return len(out_list)
 
 
+def _get_absdiff(left, right):
+
+    def make_signed(dtype):
```
Comment (klecki) [inline dali/test/python/test_utils.py:134]: Isn't it make_unsigned?

File: dali/test/python/test_utils.py (hunk 1 above)
Comment (klecki) [reply dali/test/python/test_utils.py:152]: Maybe we should explain it in a comment in the code rather than in the PR?

**Review by klecki — APPROVED** [review]

File: dali/test/python/test_utils.py (hunk 2 above)
Comment (stiepan) [reply dali/test/python/test_utils.py]: Yes it is.:)

File: dali/test/python/test_utils.py (hunk 1 above)
Comment (stiepan) [reply dali/test/python/test_utils.py:156]: done

**Comment by stiepan** [comment]
!build

**Pull request merged**